txcache/selectionTracker_test.go

```diff
@@ -0,0 +1,252 @@
+package txcache
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/stretchr/testify/require"
+)
+
+func createMockedHeaders(numOfHeaders int) []*block.Header {
+	headers := make([]*block.Header, 0)
+
+	for i := 0; i < numOfHeaders; i++ {
+		headers = append(headers, &block.Header{
+			Nonce:    uint64(i),
+			PrevHash: []byte(fmt.Sprintf("prevHash%d", i)),
+			RootHash: []byte(fmt.Sprintf("rootHash%d", i)),
+		})
+	}
+
+	return headers
+}
+
+func proposeBlocks(t *testing.T, numOfBlocks int, selectionTracker *selectionTracker, headers []*block.Header) {
```

Comment:
When a test utility is written to support testing of concurrent code, make sure to include this fact in the function's name. E.g. `proposeBlocksConcurrently`.